tests/_core/test_data.py

```diff
@@ -405,8 +405,8 @@ def test_bad_type(self, flat_list):
             PlotData(flat_list, {})
 
     @pytest.mark.skipif(
-        condition=not hasattr(pd.api, "interchange"),
-        reason="Tests behavior assuming support for dataframe interchange"
+        condition=not hasattr(pd.DataFrame, "__arrow_c_stream__"),
```

Comment:
Not sure if its a better way of expressing things but this was added in pandas 2.2

Reply:
thanks - i think we should just conditionally skip based on the pyarrow version, seeing as that's what's doing the interchanging now, that's what i've added now

Reply:
Do we need to bump the minimum version of pandas anywhere here though? Not familiar with the CI setup, but I would imagine that this PR  will not work with pandas < 2.2

Reply:
for pandas there's already an early return, so there's no issue for pandas users

https://github.com/MarcoGorelli/seaborn/blob/0bd85071284d45f38cbf419b8cf1efb2179eda24/seaborn/_core/data.py#L284-L285